CMSgov/mint-app
# Easi 2794/deescalate error logging

# EASI-2794

## Changes and Description

- Updated Splunk alerting to not alert on error = true (it leaves the responsibility of alerting to specific functions instead of to the catch all gql logging middleware
   - The main AC of the ticket was accomplished simply by updating the alert.
- Introduced a TypedError interface.
   - implemented an error code with this typedError for unauthorized errors.
   - Unwrapped the error in the GQL middleware, use zap logger to include the error code field
   - GQL errors are now logged as a warning instead of info
   



## How to test this change
1. bring up the application locally
2.  for local authentication, remove all your job codes and send a request
3.  Verify that you get a warning log in splunk for the message, that includes the error code

The main issue of de-escalataion was resolved in the SPLUNK application. If you want to verify that, you can log into splunk and verify that the alert query no longer alerts on error=true, but relies on the log level variable.




## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
